src/solve.jl

```diff
@@ -78,12 +78,10 @@ function DiffEqBase.__init(prob::DiffEqBase.AbstractDDEProblem,
   # we wrap the user-provided history function such that function calls during the setup
   # of the integrator do not fail
   if isinplace(prob)
-    ode_prob = ODEProblem{true}(ODEFunction{true}((du, u, p, t) -> f.f(du, u, h, p, t);
-                                                  mass_matrix = f.mass_matrix),
```

Comment:
We should enclose `h` here as well, otherwise function calls (e.g. when setting up a nonlinear solver) will fail.